File: populate-service/src/model/ExameSchema.java
package model;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import service.IGenericData;

public class ExameSchema implements IGenericData {

  private int id;
  private String nome;
  private String descricao;
  private List<Map<String, Object>> campos; // JSON em Java representado como lista de mapas
  private String versao;
  private List<Exame> exames = new ArrayList<>();

  public ExameSchema(int id, String nome, String descricao, List<Map<String, Object>> campos,
      String versao, List<Exame> exames) {
    this.id = id;
    this.nome = nome;
    this.descricao = descricao;
    this.campos = campos;
    this.versao = versao;
    this.exames = exames;
  }

  // Getters
  public int getId() {
    return id;
  }

  public String getNome() {
    return nome;
  }

  public String getDescricao() {
    return descricao;
  }

  public List<Map<String, Object>> getCampos() {
    return campos;
  }

  public String getVersao() {
    return versao;
  }

  public List<Exame> getExames() {
    return exames;
  }

  // Setters
  public void setNome(String nome) {
    this.nome = nome;
  }

  public void setDescricao(String descricao) {
    this.descricao = descricao;
  }

  public void setCampos(List<Map<String, Object>> campos) {
    this.campos = campos;
  }

  public void setVersao(String versao) {
    this.versao = versao;
  }

  public void setExames(List<Exame> exames) {
    this.exames = exames;
  }

  // Adicionar exame individualmente
  public void adicionarExame(Exame exame) {
    this.exames.add(exame);
  }
}

File: webapp/main/php/model/ExameSchema.php
<?php
require_once __DIR__ . '/../dao/IGenericMainData.php';
class ExameSchema implements IGenericMainData
{
    private int $id;
    private string $nome;
    private string $descricao;
    private array $campos; // vindo de um campo JSON
    private string $versao;
    private array $exames = []; // relacionamento com vários exames

    public function __construct(
        int $id,
        string $nome,
        string $descricao,
        array $campos,
        string $versao,
        array $exames = []
    ) {
        $this->id = $id;
        $this->nome = $nome;
        $this->descricao = $descricao;
        $this->campos = $campos;
        $this->versao = $versao;
        $this->exames = $exames;
    }

    // Getters
    public function getId(): int
    {
        return $this->id;
    }
    public function getNome(): string
    {
        return $this->nome;
    }
    public function getDescricao(): string
    {
        return $this->descricao;
    }
    public function getCampos(): array
    {
        return $this->campos;
    }
    public function getVersao(): string
    {
        return $this->versao;
    }
    public function getExames(): array
    {
        return $this->exames;
    }

    // Setters
    public function setNome(string $nome): void
    {
        $this->nome = $nome;
    }
    public function setDescricao(string $descricao): void
    {
        $this->descricao = $descricao;
    }
    public function setCampos(array $campos): void
    {
        $this->campos = $campos;
    }
    public function setVersao(string $versao): void
    {
        $this->versao = $versao;
    }
    public function setExames(array $exames): void
    {
        $this->exames = $exames;
    }

    // Adicionar exame individualmente
    public function adicionarExame($exame): void
    {
        $this->exames[] = $exame;
    }
}
